includes/abstracts/abstract-wc-product.php

```diff
@@ -1518,7 +1518,7 @@ public function backorders_require_notification() {
 	 * @return bool
 	 */
 	public function is_on_backorder( $qty_in_cart = 0 ) {
-		return $this->managing_stock() && $this->backorders_allowed() && ( $this->get_stock_quantity() - $qty_in_cart ) < 0 ? true : false;
+		return $this->managing_stock() && $this->backorders_allowed() && ( $this->get_stock_quantity() - $qty_in_cart );
```

Comment:
This may introduced a bug... Should keep `< 0`.

This is how works today:

```
wp> (1) < 0
=> bool(false)
wp> (0) < 0
=> bool(false)
wp> (-1) < 0
=> bool(true)
```

And this after your change:

```
wp> (bool) 1
=> bool(true)
wp> (bool) 0
=> bool(false)
wp> (bool) -1
=> bool(true)
```